cvmfs/sync_item.cc

```diff
@@ -10,11 +10,86 @@
 
 using namespace std;  // NOLINT
 
+
+static bool ProcessCatalogProperties(const std::string &scratch_path,
+                                     bool &external_data) {  // NOLINT
+  char buf[128];
+  external_data = false;
+  ssize_t retval;
+  if (-1 != (retval = platform_lgetxattr(scratch_path.c_str(),
```

Comment:
I wonder if the extended attribute is necessary at all.  Can't we just look inside the catalog marker if the file size is non-zero?